configure.ac

```diff
@@ -676,33 +676,54 @@ if test x"${with_d}" = xno; then
   AC_MSG_NOTICE([Disabling D])
   D2COMPILER=
 else
-  old_ac_ext=$ac_ext
-  ac_ext=d
 
   if test -z "$D2COMPILERBIN" ; then
-    AC_CHECK_PROGS(D2COMPILER, dmd gdmd ldmd2 ldc2)
-
+    AC_CHECK_PROGS(D2COMPILER, dmd ldmd2 ldc2 gdmd gdc)
     if test -n "$D2COMPILER" ; then
+      old_ac_ext=$ac_ext
+      ac_ext=d
+      case $D2COMPILER in
+        gdc)
+          d_out=-oconftest.out
+          D2COMPILER_TYPE=GCC
+          ;;
+        ldc2|ldmd2)
+          d_out=-ofconftest.out
+          D2COMPILER_TYPE=LLVM
+          ;;
+        *)
+          d_out=-ofconftest.out
+          D2COMPILER_TYPE=DMD
+          ;;
+      esac
```

Comment:
Wondering how `gdmd` should be handled here, I looked up what it was and found:

```
$ apt show gdmd
Package: gdmd
Version: 2.100+git210818-1
Priority: optional
Section: devel
Maintainer: Debian D Language Group <team+d-team@tracker.debian.org>
Installed-Size: 38.9 kB
Depends: gdc (>= 4:12.2)
Homepage: https://github.com/D-Programming-GDC/gdmd
Download-Size: 9,856 B
APT-Sources: http://deb.debian.org/debian unstable/main amd64 Packages
Description: DMD-like wrapper for GDC
 This package provides `gdmd`, an interface to the GNU D compiler (GDC), which
 recognizes flags normally used with the DMD D compiler and translates them to
 flags GDC can understand.
 This makes `gdmd` a drop-in replacement to the `dmd` compiler invocation.
 .
 The GNU D compiler compiles D sources using the gcc backend to generate
 highly optimized binaries.
```

So we could just install `gdmd` instead of `gdc` for CI (and tell people they need it if they want to run SWIG/D's testsuite locally with gdc) and avoid having to special-case the command line options.

Reply:
OK, I've read the rest of the patch now.

I see there are CI jobs for both `gdc` and `gdmd`, so I would suggest we drop that `gdc` job and the special handling for the command line options.

Reply:
I do not see it as a big overhead, but I do understand the need to reduce CI.
Your call.

Reply:
Is it OK now?

Reply:
> Perhaps we need a more flexible CI.
> One that run a minimal mandatory set and add more tests for the specific language/s that have a change.

I've suggested that before, but wsfulton didn't seem keen.

Reply:
> I do not see it as a big overhead, but I do understand the need to reduce CI.

I mean sure it's only 1% more, but that argument can be used for each additional CI job and that won't end well.  Testing both gdc and gdmd doesn't seem to add useful test coverage (because gdmd is just a command line option translation wrapper) and if we just go with gdmd we can eliminate the need to handle two different command line syntaxes.